src/common/feedGenerator.ts

```diff
@@ -56,31 +57,52 @@ export const whereKeyword = (
   return `EXISTS${query}`;
 };
 
+export const getExcludedAdvancedSettings = async (
+  con: ORMConnection,
+  feedId: string,
+  userId: string,
+): Promise<number[]> => {
+  const [features, advancedSettings, feedAdvancedSettings] = await Promise.all([
+    fetchUserFeatures(userId),
+    con.getRepository(AdvancedSettings).find(),
+    con.getRepository(FeedAdvancedSettings).find({ feedId, enabled: false }),
+  ]);
+  const userSettings = feedAdvancedSettings.reduce(
+    (obj, settings) => ({
+      ...obj,
+      [settings.advancedSettingsId]: settings.enabled,
+    }),
+    {},
+  );
+  const feature = features?.advanced_settings_default_values;
+  const excludedSettings = advancedSettings.filter((adv) => {
+    const hasNoSettings = userSettings[adv.id] === undefined;
+    const conditions = [
+      userSettings[adv.id],
+      feature?.value[adv.id],
+      !(hasNoSettings && adv.defaultEnabledState === false),
+    ];
+
+    return conditions.some((value) => value === false);
```

Comment:
Is this the correct assumption?

- default: false
- feed: false
- user: true

The whole thing will return false?
Or does that not matter for this part?

Reply:
That should be the expected result, if there is any false it should automatically be included in the list.

Reply:
Oke great 👏